doc/modules/preprocessing.rst

```diff
@@ -593,6 +593,51 @@ the 2 features with respectively 3 categories each.
 See :ref:`dict_feature_extraction` for categorical features that are
 represented as a dict, not as scalars.
 
+.. _target_regressor_encoder:
+
+Target Regressor Encoder
+========================
+The :class:`~sklearn.preprocessing.TargetRegressorEncoder` uses statistics of
```

Comment:
We should start be describe in which cases this Encoder could or should be used. AFAIK it's useful when there are lots of unordered categories, so the OHE would be too expensive? Zipcodes sounds like the go-to illustration use-case?